File: python/svix/api/endpoint_autoconfig.py
# This file is @generated
import typing as t
from dataclasses import dataclass

from ..models import AutoConfigOut, RotateSubscriptionIn2
from .common import ApiBaseAsync, ApiBaseSync, BaseOptions, serialize_params


@dataclass
class EndpointAutoconfigCreateOptions(BaseOptions):
    idempotency_key: t.Optional[str] = None

    def _header_params(self) -> t.Dict[str, str]:
        return serialize_params(
            {
                "idempotency-key": self.idempotency_key,
            }
        )


@dataclass
class EndpointAutoconfigRotateOptions(BaseOptions):
    idempotency_key: t.Optional[str] = None

    def _header_params(self) -> t.Dict[str, str]:
        return serialize_params(
            {
                "idempotency-key": self.idempotency_key,
            }
        )


class EndpointAutoconfigAsync(ApiBaseAsync):
    async def create(
        self,
        app_id: str,
        options: EndpointAutoconfigCreateOptions = (EndpointAutoconfigCreateOptions()),
    ) -> AutoConfigOut:
        """Create an AutoConfig subscription."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/app/{app_id}/autoconfig",
            path_params={
                "app_id": app_id,
            },
            query_params=options._query_params(),
            header_params=options._header_params(),
        )
        return AutoConfigOut.model_validate(response.json())

    async def rotate(
        self,
        app_id: str,
        autoconfig_id: str,
        rotate_subscription_in2: RotateSubscriptionIn2,
        options: EndpointAutoconfigRotateOptions = (EndpointAutoconfigRotateOptions()),
    ) -> AutoConfigOut:
        """Rotate the auth token and signing secret for an AutoConfig subscription."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}/rotate",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
            query_params=options._query_params(),
            header_params=options._header_params(),
            json_body=rotate_subscription_in2.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return AutoConfigOut.model_validate(response.json())


class EndpointAutoconfig(ApiBaseSync):
    def create(
        self,
        app_id: str,
        options: EndpointAutoconfigCreateOptions = (EndpointAutoconfigCreateOptions()),
    ) -> AutoConfigOut:
        """Create an AutoConfig subscription."""
        response = self._request_sync(
            method="post",
            path="/api/v1/app/{app_id}/autoconfig",
            path_params={
                "app_id": app_id,
            },
            query_params=options._query_params(),
            header_params=options._header_params(),
        )
        return AutoConfigOut.model_validate(response.json())

    def rotate(
        self,
        app_id: str,
        autoconfig_id: str,
        rotate_subscription_in2: RotateSubscriptionIn2,
        options: EndpointAutoconfigRotateOptions = (EndpointAutoconfigRotateOptions()),
    ) -> AutoConfigOut:
        """Rotate the auth token and signing secret for an AutoConfig subscription."""
        response = self._request_sync(
            method="post",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}/rotate",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
            query_params=options._query_params(),
            header_params=options._header_params(),
            json_body=rotate_subscription_in2.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return AutoConfigOut.model_validate(response.json())

File: php/src/Api/EndpointAutoconfig.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Api;

use Svix\Exception\ApiException;
use Svix\Models\AutoConfigOut;
use Svix\Models\RotateSubscriptionIn2;
use Svix\Request\SvixHttpClient;

class EndpointAutoconfig
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /**
     * Create an AutoConfig subscription.
     *
     * @throws ApiException
     */
    public function create(
        string $appId,
        EndpointAutoconfigCreateOptions $options = new EndpointAutoconfigCreateOptions(),
    ): AutoConfigOut {
        $request = $this->client->newReq('POST', "/api/v1/app/{$appId}/autoconfig");
        $request->setHeaderParam('idempotency-key', $options->idempotencyKey);
        $res = $this->client->send($request);

        return AutoConfigOut::fromJson($res);
    }

    /**
     * Rotate the auth token and signing secret for an AutoConfig subscription.
     *
     * @throws ApiException
     */
    public function rotate(
        string $appId,
        string $autoconfigId,
        RotateSubscriptionIn2 $rotateSubscriptionIn2,
        EndpointAutoconfigRotateOptions $options = new EndpointAutoconfigRotateOptions(),
    ): AutoConfigOut {
        $request = $this->client->newReq('POST', "/api/v1/app/{$appId}/autoconfig/{$autoconfigId}/rotate");
        $request->setHeaderParam('idempotency-key', $options->idempotencyKey);
        $request->setBody(json_encode($rotateSubscriptionIn2));
        $res = $this->client->send($request);

        return AutoConfigOut::fromJson($res);
    }
}
